src/common/components/fileUpload/FileUpload.test.tsx

```diff
@@ -64,29 +74,83 @@ test('Should show amount of successful files uploaded correctly when file fails
       multiple
       uploadFunction={uploadFunction}
       fileDeleteFunction={() => Promise.resolve()}
+      existingAttachments={[existingFileA, existingFileB]}
     />,
     undefined,
     undefined,
     { applyAccept: false },
   );
   const fileUpload = screen.getByLabelText(inputLabel);
   user.upload(fileUpload, [
-    new File(['test-a'], 'test-file-a.png', { type: 'image/png' }),
-    new File(['test-b'], 'test-file-b.pdf', { type: 'application/pdf' }),
+    new File(['test-a'], fileNameA, { type: 'image/png' }),
+    new File(['test-b'], fileNameB, { type: 'application/pdf' }),
+    new File(['test-c'], fileNameC, { type: 'image/png' }),
   ]);
 
   await waitFor(() => {
-    expect(screen.queryByText('1/2 tiedosto(a) tallennettu')).toBeInTheDocument();
+    expect(screen.queryByText('1/3 tiedosto(a) tallennettu')).toBeInTheDocument();
   });
+  expect(
+    screen.queryByText('Liitteen tallennus epäonnistui 2/3 tiedostolle', { exact: false }),
+  ).toBeInTheDocument();
+  expect(
+    screen.queryByText(
+      `Tiedosto (${fileNameB}) on viallinen. Tarkistathan, että tiedostomuoto on oikea eikä sen koko ylitä sallittua maksimikokoa.`,
+      { exact: false },
+    ),
+  ).toBeInTheDocument();
+  expect(
+    screen.queryByText(
+      `Valittu tiedostonimi (${fileNameA}) on jo käytössä. Nimeä tiedosto uudelleen.`,
+      { exact: false },
+    ),
+  ).toBeInTheDocument();
 });
 
-test('Should show amount of successful files uploaded correctly when request fails', async () => {
+test('Should show amount of successful files uploaded and errors correctly when request fails for bad request', async () => {
   server.use(
     rest.post('/api/hakemukset/:id/liitteet', async (req, res, ctx) => {
       return res(ctx.status(400), ctx.json({ errorMessage: 'Failed for testing purposes' }));
     }),
   );
 
+  const fileNameA = 'test-file-a.png';
+  const inputLabel = 'Choose a file';
+  const { user } = render(
```

Comment:
Can the rendered FileUpload component be separated to some sort of a helper function to get rid of the duplication warnings in sonar? The function could take some input arguments and / or use some default values if different data is needed.